ibis/expr/rules.py

```diff
@@ -452,22 +452,23 @@ def output_type(self):
     return output_type
 
 
-def numeric_highest_promote(i):
+# TODO: UNUSED
+# def numeric_highest_promote(i):
```

Comment:
If things are unused, you can delete them. We can always resurrect them later with git if needed.